File: app/src/main/java/com/example/instabugsearchwords/presentation/WordsListViewModelFactory.java
package com.example.instabugsearchwords.presentation;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.example.instabugsearchwords.domain.get_words.GetWordsUseCase;

public class WordsListViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    GetWordsUseCase getWordsUseCase;

    public WordsListViewModelFactory(GetWordsUseCase getWordsUseCase) {
        this.getWordsUseCase = getWordsUseCase;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if(modelClass.isAssignableFrom(WordsListViewModel.class)){
            return (T) new WordsListViewModel(getWordsUseCase);
        }
        throw new IllegalArgumentException("Unknown view model found");
    }
}

File: app/src/main/java/com/example/instabugsearchwords/presentation/WordsListViewModelFactory.kt
package com.example.instabugsearchwords.presentation

import com.example.instabugsearchwords.domain.get_words.GetWordsUseCase
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.example.instabugsearchwords.presentation.WordsListViewModel
import java.lang.IllegalArgumentException

class WordsListViewModelFactory(var getWordsUseCase: GetWordsUseCase) : ViewModelProvider.Factory {

    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(WordsListViewModel::class.java)) {
            return WordsListViewModel(getWordsUseCase) as T
        }
        throw IllegalArgumentException("Unknown view model found")
    }
}